L/libclangex/build_tarballs.jl

```diff
@@ -85,10 +107,10 @@ non_reg_ARGS = filter(arg -> arg != "--register", non_platform_ARGS)
 
 for (i,build) in enumerate(builds)
     build_tarballs(i == lastindex(builds) ? non_platform_ARGS : non_reg_ARGS,
-                   name, version, sources, build.script,
+                   name, version, build.sources, script,
                    build.platforms, build.products, build.dependencies;
-                   preferred_gcc_version=v"8", julia_compat="1.7",
-                   augment_platform_block)
+                   preferred_gcc_version=v"12", julia_compat="1.12",
```

Comment:
Is the bump to `v"12"` needed for ClangExtra functionality? LLVM 18 only needs GCC 10.